src/web/app/components/question-types/question-constraint/contribution-question-constraint.component.ts

```diff
@@ -9,13 +18,102 @@ import { QuestionConstraintComponent } from './question-constraint.component';
   templateUrl: './contribution-question-constraint.component.html',
   styleUrls: ['./contribution-question-constraint.component.scss'],
 })
-export class ContributionQuestionConstraintComponent extends QuestionConstraintComponent<any> {
+export class ContributionQuestionConstraintComponent
+    extends QuestionConstraintComponent<FeedbackContributionQuestionDetails> {
 
   constructor() {
-    super(null);
+    super(DEFAULT_CONTRIBUTION_QUESTION_DETAILS());
+  }
+
+  /**
+   * Checks if all forms are answered.
+   */
+  get isAllFormsAnswered(): boolean {
```

Comment:
The comments in these file are probably not necessary as the method names are self-explanatory. See https://teammates.github.io/teammates/best-practices/coding.html#code-comments.